File: Frontend/src/types/record.ts
export interface Record {
    recordId: number
    username: string
    bookName: string
    borrowDate: Date
    DueDate: Date
    returnDate: Date
    status: number
}

export interface RecordRoot {
    records: Record[]
}

export interface Detail {
    recordId: number
    username: string
    email: string,
    avatar: string,
    bookName: string,
    description: string,
    author: string,
    cover: string,
    tags: string[],
    borrowDate: Date
    DueDate: Date
    returnDate: Date
    status: number
}

export interface DetailRoot {
    details: Detail[]
}

export interface userRecord {
    recordId: number
    username: string
    bookName: string,
    author: string,
    cover: string,
    tags: string[],
    borrowDate: Date
    DueDate: Date
    returnDate: Date
    status: number
}

export interface userRecordRoot {
    userRecords: userRecord[]
}

File: backend/src/main/java/com/readnest/pojo/Record.java
package com.readnest.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Record {
    private Integer recordId;
    private String username;
    private String bookName;
    private Date borrowDate;
    private Date dueDate;
    private Date returnDate;
    private Integer status;
}
